modules/debugging/index.qmd

```diff
@@ -95,29 +94,271 @@ detach_package(tidyverse)
+:::
+
+The process of cleaning up your code and reducing it down into a small example is another great exercise that often helps you identify the problem and fix it yourself.
+
+Remember, you need to provide **all code necessary** to **load in packages you use**, **read in data** and **run functions** that cause the error.
+
+## Provide session information
+
+Sometimes the error is specific to the version of R, or the packages you are using. To help others help you, you can provide the session information using the `sessioninfo::sessionInfo()` function. This will print out the version of R you are using, and the versions of the packages you have loaded, which makes it easier to fix some version, language or OS specific problems. A good reproducible example should come with this session information output.
+
+## Using the `reprex` package for producing examples
+
+the [`reprex`](https://reprex.tidyverse.org/) package is a useful tool that can help you make a reproducible example. It can be used to run the code in your example in a clean environment, which is a good way to test if you've included everything necessary for others to reproduce the error. It'll also copy a neatly formatted version of your example, which includes images and console output from the example as it runs on your computer. It can even include a collapsible summary of your session information.
+
+-   add images
+-   potentially a video here
+
+# Asking for help checklist ☑️
+
+This is **not an exhaustive check-list.** What needs to be included and checked is very question specific, but this checklist should apply to most questions.
+
+-   [ ] Is the problem [clearly]{.underline} and [succinctly described]{.underline}?
+-   [ ] Is the [expected solution or behaviour]{.underline} outlined?
+-   [ ] Has [nobody asked this question]{.underline} before? (Search the web!)
+-   [ ] Are you asking the [right people]{.underline} at the [right place]{.underline}?
+-   [ ] Have you added [appropriate tags]{.underline} or [keywords]{.underline} to reach the right experts?
+-   [ ] Have you prepared your example using the [reprex package]{.underline} to ensure the problem is easily reproduced on other computers?
+-   [ ] Have you [minimised your problem]{.underline} as much as possible, [removing irrelevant packages, data and code]{.underline}?
+
+**If your questions involve a dataset**
+
+-   [ ] Is the data supplied
+-   [ ] If the data is big, could you cull your data further to communicate or reproduce the problem?
+
+# Asking for help checklist ☑️
+
+This is **not an exhaustive check-list.** What needs to be included and checked is very question specific, but this checklist should apply to most questions.
+
+-   [ ] Is the problem [**clearly**]{style="background-color: yellow;"} and [**succinctly described**]{style="background-color: yellow;"}?
```

Comment:
We can add CSS for highlight for consistency across articles, [text]{.highlight} could be nicer?